### ISSUE TITLE
Validate Fisher results 

### ISSUE THREAD

**Issue by dlanzieri** [LSSTDESC/DifferentiableHOS#16]
This issue is to track the steps we need compute in order to be sure that we can trust our Fisher matrix.
As discussed before, the steps to follow are: 

- [ ] Determine how many samples we need to be able to trust our Fisher matrix
- [ ] As soon as we trust the matrix, see what the contours look like for PS and peaks when all params are included
- [ ] See what the contours look like with just Omega_c and sigma8
- [ ] Based on the previous results, understand which degree of freedom  leads to an important change in the behaviour of the constraints in Omega_c and sigma8

**Comment by dlanzieri** [comment]
@EiffL Concerning the first task, the following image shows the overlapped Fisher contours made by averaging over a different number of samples.
The red contours are realized using all the samples (170 jacobians), the violet contours just 34. 
What I see from this plot it's that the contours became larger when I use a larger sample for the average.
![all](https://user-images.githubusercontent.com/72620117/141298314-ddfa921a-a505-41b4-9152-63f53cb79804.png)
In the following plot I'm only showing the contours realized with :
RED: 170 jacobians
YELLOW : 136 jacobians
GREEN: 102 jacobians
![last3](https://user-images.githubusercontent.com/72620117/141298326-0076c107-6cba-4678-8f3a-3c4662ba9646.png)
It seems that between 102 and 136 samples the contours are still differents, but between 136 and 170, they start to be more similars and stables. 
Here, I just keep:
RED: 170 jacobians
YELLOW : 136 jacobians
![last2](https://user-images.githubusercontent.com/72620117/141298333-a130cd3f-7fbe-4bb0-83fb-282fe614a71f.png)

**Comment by dlanzieri** [comment]
@EiffL This is what I was thinking to do:  I would increase a bit the sample used to compute the average (at least for the power spectrum and in this validation stage). Then, If we see that this trend is confirmed and that with ~180/200 objects we can trust of our Fisher matrix, I will use this number for the further analysis (peak counts, l1norm, IA etc). 
What do you think?

**Comment by EiffL** [comment]
Ah very nice! To make it a round number we could say 200 samples? to ensure the jacobian is stable.

**Comment by EiffL** [comment]
Note that you would then want to do a similar check for your other stastistics. Because it all depends on how noisy a  given bin is. Hopefully 200 will be a good number for  all of them

**Comment by dlanzieri** [comment]
@EiffL I increased the number of samples to average the Ps-Jacobian. 
I made a plot of the mean Ps-Jacobian computed  with different numbers of samples (normalized by the square root of the diagonal of the covariance matrix). You can find in this [ notebook](https://github.com/LSSTDESC/DifferentiableHOS/blob/7e928fe541ae9fe9de4279c499df60de2b20da74/notebooks/Fisher_tuning.ipynb) the plots for more different number of samples, for an easier visualization here I show only 3 comparisons. 
The Jacobians seem still noisy, especially for sigma8, h and ns. 
![jac_ps](https://user-images.githubusercontent.com/72620117/141499040-ca074e97-430c-4304-9357-0ef10ba53d34.png)
and in the following plot I'm only showing the contours realized with :
RED: 250 jacobians
Violet: 230 jacobians
Orange: 201 jacobians
![last_ps](https://user-images.githubusercontent.com/72620117/141499048-29bf5ec1-79ba-4ff1-b988-b223ca27237e.png)
Then, I made a similar check also for the peak counts. 
Fortunately, the jacobians seem less noisy and much more stables (slightly noisy h and ns)
![jac_pcounts](https://user-images.githubusercontent.com/72620117/141499054-03c8667e-9f52-4c43-bde5-5755007401d9.png)
and in the following plot I'm only showing the contours realized with :
RED: 160 jacobians
Blue : 230 jacobians
Yellow: 200 jacobians
![all_pk](https://user-images.githubusercontent.com/72620117/141499061-29106849-0c5d-42b3-bcc4-2847c41d23a6.png)
For both the statistics the contours don't seem change their orientations too much with a number of samples higher than ~160 (in the notebook  you can find  other 2 plots with more number of sample, and there the change of orientation is more evident).
However, the width of the contours change for both statistics with a different number of samples.

**Comment by EiffL** [comment]
This is great Denise! So, yeah Fisher matrices are always tricky :-/ but I think this is not too bad. It looks "fairly stable" and we have an idea of how much of an error to expect on the contours, so we know if when comparing 2 different stats we should read too much into some differences.

**Comment by dlanzieri** [comment]
@EiffL We have also discussed the importance of making the power spectrum analysis consistent with the peak counts analysis in order to compare their sensitivity.
If we apply 1 arcmin smoothing the plot of the angular power spectrum looks like this:
![ps_comp_1](https://user-images.githubusercontent.com/72620117/141775090-9471fda8-8354-4a39-b444-f977362e8af5.png)
As we already noted, the effect of smoothing starts to suppress the information for ell around 10000, but we consider the power spectrum only in a range between 300 and 3000. 
If we wanted to apply a smoothing so that the angular power spectrum lost power for ell = 3000, we should apply a higher smoothing. This for example corresponds to 5 arcmin:
![ps_comp_5](https://user-images.githubusercontent.com/72620117/141775104-07787387-ec49-4959-a3c7-d83c0d2d85f0.png)
 I was wondering if we should increase the smoothing or consider the power spectrum until ell~ 10000.

**Comment by EiffL** [comment]
You are correct, this 5 arcmin smoothing seems more consistent. For an LSST Y1 this is probably around the scales we would use. 
In recent DES papers they went to 7.5 arcmin https://arxiv.org/pdf/2110.10135.pdf

**Comment by EiffL** [comment]
hummm but can you convert the l=3000 to an arcmin scale? what value do you get?

**Comment by dlanzieri** [comment]
l=3000 should be ~ 3.6
![ps_comp_cor](https://user-images.githubusercontent.com/72620117/141797885-9538c1d3-57de-4929-82a1-ee428024f9f0.png)
